skorch/dataset.py

```diff
@@ -281,3 +281,23 @@ def __call__(self, dataset, y=None, groups=None):
     def __repr__(self):
         # pylint: disable=useless-super-delegation
         return super(CVSplit, self).__repr__()
+
+
+class TrainValidDataset:
+    """Small wrapper class to support custom training and validation
```

Comment:
Here and further down, you describe this as "custom datasets" but this is misleading. A custom dataset suggests that I implemented my own type of dataset. I would rather go with "predefined" or "pre-existing split" (maybe the latter to avoid confusion with sklearn's `PredefinedSplit`). Also, I would add a newline after the first sentence.